src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredToArmProcessorImpl.java

```diff
@@ -0,0 +1,194 @@
+
+        ExternalObjectDirectoryEntity  armExternalObjectDirectoryEntity = new ExternalObjectDirectoryEntity();
+        armExternalObjectDirectoryEntity.setExternalLocationType(externalLocationTypeRepository.getReferenceById(ARM.getId()));
+        armExternalObjectDirectoryEntity.setStatus(objectDirectoryStatusRepository.getReferenceById(ARM_INGESTION.getId()));
+        armExternalObjectDirectoryEntity.setExternalLocation(externalObjectDirectory.getExternalLocation());
+
+        MediaEntity mediaEntity = externalObjectDirectory.getMedia();
+        if (mediaEntity != null) {
+            armExternalObjectDirectoryEntity.setMedia(mediaEntity);
+        }
+
+        TranscriptionDocumentEntity transcriptionDocumentEntity = externalObjectDirectory.getTranscriptionDocumentEntity();
+        if (transcriptionDocumentEntity != null) {
+            armExternalObjectDirectoryEntity.setTranscriptionDocumentEntity(transcriptionDocumentEntity);
+        }
+
+        AnnotationDocumentEntity annotationDocumentEntity = externalObjectDirectory.getAnnotationDocumentEntity();
+        if (annotationDocumentEntity != null) {
+            armExternalObjectDirectoryEntity.setAnnotationDocumentEntity(annotationDocumentEntity);
+        }
+        OffsetDateTime now = OffsetDateTime.now();
+        armExternalObjectDirectoryEntity.setCreatedDateTime(now);
+        armExternalObjectDirectoryEntity.setLastModifiedDateTime(now);
+        var systemUser = userAccountRepository.getReferenceById(SystemUsersEnum.DEFAULT.getId());
+        armExternalObjectDirectoryEntity.setCreatedBy(systemUser);
+        armExternalObjectDirectoryEntity.setLastModifiedBy(systemUser);
+        armExternalObjectDirectoryEntity.setTransferAttempts(1);
+
+        return armExternalObjectDirectoryEntity;
+    }
+
+
+    public String generateFilename(ExternalObjectDirectoryEntity externalObjectDirectoryEntity) {
+        final Integer entityId = externalObjectDirectoryEntity.getId();
+        final Integer transferAttempts = externalObjectDirectoryEntity.getTransferAttempts();
+
+        Integer documentId = 0;
+        if (externalObjectDirectoryEntity.getMedia() != null) {
+            documentId = externalObjectDirectoryEntity.getMedia().getId();
+        }
+        if (externalObjectDirectoryEntity.getTranscriptionDocumentEntity() != null) {
+            documentId = externalObjectDirectoryEntity.getTranscriptionDocumentEntity().getId();
+        }
+        if (externalObjectDirectoryEntity.getAnnotationDocumentEntity() != null) {
+            documentId = externalObjectDirectoryEntity.getAnnotationDocumentEntity().getId();
+        }
+
+        return String.format("%s_%s_%s", entityId, documentId, transferAttempts);
+    }
+
+    private void updateTransferAttempts(ExternalObjectDirectoryEntity externalObjectDirectoryEntity) {
+        int currentNumberOfAttempts = externalObjectDirectoryEntity.getTransferAttempts();
```

Comment:
Looking at staging this value seems to be set as null in some cases. Might be worth a null check